rplugin/python3/deoplete/deoplete.py

```diff
@@ -235,108 +244,100 @@ def profile_end(self, name):
-            source.max_menu_width = getattr(
-                source, 'max_menu_width',
-                context['vars']['deoplete#max_menu_width'])
-            if hasattr(source, 'on_init'):
-                self.debug('on_init Source: %s (%s)', source.name)
-                source.on_init(context)
-
-            self.__sources[source.name] = source
-            self.debug('Loaded Source: %s (%s)',
-                       source.name, module.__file__)
+            source = None
+
+            try:
+                Source = import_plugin(path, 'source', 'Source')
+                if Source is None:
+                    continue
+                source = Source(self.__vim)
+                source.name = getattr(source, 'name', name)
+
+                source.min_pattern_length = getattr(
+                    source, 'min_pattern_length',
+                    context['vars']['deoplete#auto_complete_start_length'])
+                source.max_abbr_width = getattr(
+                    source, 'max_abbr_width',
+                    context['vars']['deoplete#max_abbr_width'])
+                source.max_menu_width = getattr(
+                    source, 'max_menu_width',
+                    context['vars']['deoplete#max_menu_width'])
+
+                if hasattr(source, 'on_init'):
+                    self.debug('on_init Source: %s', source.name)
+                    source.on_init(context)
+            except Exception:
+                error_tb(self.__vim, 'Could not load source: %s' % name)
+            finally:
+                if source is not None:
+                    self.__sources[source.name] = source
+                    self.debug('Loaded Source: %s (%s)', source.name, path)
 
         self.set_source_attributes(context)
         self.__custom = context['custom']
-        # self.debug(self.__sources)
 
     def load_filters(self, context):
         # Load filters from runtimepath
-        for path in globruntime(context['runtimepath'],
-                                'rplugin/python3/deoplete/filter/base.py'
-                                ) + globruntime(
-                                    context['runtimepath'],
-                                    'rplugin/python3/deoplete/filter/*.py'):
-            filename = os.path.basename(path)[: -3]
-            module = importlib.machinery.SourceFileLoader(
-                'deoplete.filter.' + filename, path).load_module()
-            if hasattr(module, 'Filter') and filename not in self.__filters:
-                filter = module.Filter(self.__vim)
-                self.__filters[filter.name] = filter
-                self.debug('Loaded Filter: %s (%s)',
-                           filter.name, module.__file__)
-        # self.debug(self.__filters)
+        for path in find_rplugins(context, 'filter'):
+            name = os.path.splitext(os.path.basename(path))[0]
+            if name in self.__filters:
+                continue
+
+            filter_ = None
```

Comment:
Ah, I thought I changed that before I committed.  It's something that's done in Python's standard library when a name conflicts with a builtin/keyword.  Like `class_`.

I'll change it.

This reminds me of another problem when I was trying to look for a simple solution to #327.  The use of the `__` for the "private" variables is considered to be internal magic like `__str__()` or `__eq__()`.  It doesn't need to be changed, but it made it impossible for me to check `hasattr(self, '__vim')` to figure out the logging issue.